plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,31 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void get_time_format_string(char *str, size_t maxsize,
+                      const char *time_key_format, const struct tm *timeptr,
+                      size_t len, struct flb_time *tms){
+    size_t s;
+    const char *format = "%Y-%m-%dT%H:%M:%S";
+    len = strftime(str, maxsize, format, timeptr);
+    format = strstr(time_key_format, ".%F");
+    if (format) {
+        s = snprintf(str + len, sizeof(str) - 1 - len,
+                     ".%03" PRIu64 "Z",
+                     (uint64_t) tms->tm.tv_nsec / 1000000);
+    } else {
+        format = strstr(time_key_format, ".%L");
+        if (format) {
+            s = snprintf(str + len, sizeof(str) - 1 - len,
```

Comment:
So len is the number of bytes that strftime wrote... if I am understanding this correctly, then it is just appending the nanoseconds or milliseconds at the end and also a "Z" character. Will it always add a "Z" at the end? 

Can this code handle timestamps where there is no Z at the end or where the nanoseconds are not at the end? For example. what if the timestamp is backwards to our normal order, with nanosecond, then second, then minute, etc